src/jit/codegenxarch.cpp

```diff
@@ -7397,30 +7397,106 @@ void CodeGen::genPutArgStk(GenTreePtr treeNode)
     }
     else if (data->OperGet() == GT_FIELD_LIST)
     {
-        GenTreeFieldList* fieldList = data->AsFieldList();
-        // TODO-X86-CQ: If there are no gaps or padding in the layout, we can use a series of pushes.
-        // For now, always make space for the struct, and store each field at its offset.
+        GenTreePutArgStk* const putArgStk = treeNode->AsPutArgStk();
```

Comment:
It might be reasonable to change the signature of this method to take a GenTreePutArgStk*.